### PR TITLE
Add class-level comments to version-related classes

### PR DESCRIPTION
**Pull request by riversand963**
As title.

### CODE REVIEW

**Comment by miasantreble** [comment]
Is there plan to add (or existing) comment about the relationship between `Version`, `VersionSet`, `VersionEdit`, `VersionStorageInfo`?

**Comment by riversand963** [comment]
> Is there plan to add (or existing) comment about the relationship between `Version`, `VersionSet`, `VersionEdit`, `VersionStorageInfo`?

VersionEdit is in another file, thus won't be documented here. I think I have included a high-level description of Version, VersionStorageInfo and VersionSet before each of the classes. Are you suggesting a file level explanation?

**Comment by miasantreble** [comment]
> Are you suggesting a file level explanation?
Something to describe the relationship between these classes. When reading RocksDB internal code, I often wonder how are these classes related

File: db/version_set.h [hunk 1]
```diff
@@ -747,6 +752,8 @@ struct ObsoleteFileInfo {
 
 class BaseReferencedVersionBuilder;
 
+// VersionSet is the collection of versions of all the column families of the
+// database. Each database owns one VersionSet.
```
Comment (siying) [inline db/version_set.h]: This is good, but can you also explain how column family is related to this data structure?

Comment (riversand963) [reply db/version_set.h]: Sure, will update.

**Pull request closed without merging**